distribution/pull_v2.go

```diff
@@ -174,90 +179,101 @@ func (ld *v2LayerDescriptor) Download(ctx context.Context, progressOutput progre
 			if err := os.Remove(ld.tmpFile.Name()); err != nil {
 				logrus.Errorf("Failed to remove temp file: %s", ld.tmpFile.Name())
 			}
-			ld.tmpFile, err = createDownloadFile()
+			ld.tmpFile, err = createDownloadFile(cacheDir)
 			if err != nil {
 				return nil, 0, xfer.DoNotRetry{Err: err}
 			}
-		} else if offset != 0 {
-			logrus.Debugf("attempting to resume download of %q from %d bytes", ld.digest, offset)
 		}
 	}
+	if offset != 0 {
+		logrus.Debugf("attempting to resume download of %q from %d bytes", ld.digest, offset)
+	}
 
 	tmpFile := ld.tmpFile
 
-	layerDownload, err := ld.open(ctx)
-	if err != nil {
-		logrus.Errorf("Error initiating layer download: %v", err)
-		return nil, 0, retryOnError(err)
-	}
-
-	if offset != 0 {
-		_, err := layerDownload.Seek(offset, os.SEEK_SET)
+	if ld.verifier == nil || !ld.verifier.Verified() {
```

Comment:
Can you please factor the body of this `if` statement into a separate function? I think that will make it easier to read.